org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterSwitchLevel.java

```diff
@@ -125,11 +168,19 @@ public void attributeUpdated(ZclAttribute attribute) {
         logger.debug("{}: ZigBee attribute reports {}", endpoint.getIeeeAddress(), attribute);
         if (attribute.getCluster() == ZclClusterType.LEVEL_CONTROL
                 && attribute.getId() == ZclLevelControlCluster.ATTR_CURRENTLEVEL) {
-            Integer level = (Integer) attribute.getLastValue();
-            if (level != null) {
-                updateChannelState(levelToPercent(level));
+            lastLevel = levelToPercent((Integer) attribute.getLastValue());
```

Comment:
This change here removes the non-null check for the last level. Is this intended, or is such a check still needed to avoid NPEs, e.g., when calling `levelToPercent`?

Reply:
Yes - it was intended as I can't see any way that it should get to here and be null. Given the notification is for an update of this specific attribute, it should not be null and I considered this should be ok to remove.

Reply:
OK, thanks!